### DISCUSSION TITLE
Is there a way to learn by imitation?

### DISCUSSION THREAD

**Discussion by tkzcxl** [asieradzk/RL_Matrix#6] [Ideas]
Thank you for providing this great tool. I want to try using it to train game AI from scratch, which might be a bit slow. If I could learn by imitation using expert data, that would be great (like the functionality of Unity's ML-Agent)

**Comment by tkzcxl** [comment]
Oh, I noticed the roadmap, but after successfully trying the example CartPole, I ran into a new issue, I could only load/save.pt models and didn't find a way to store as onnx models, what do I do if I want to embed in distributed games/software and carrying a huge torch library is not ideal?

**Reply by asieradzk** [reply]
Hey. Thank you for trying RL Matrix.

I did make some basic behavioural cloning without high quality code. Same for GAIL. It was working poorly for my project so I kinda let it rest for a while, but chances are I will try again soon. Let's keep in touch so I can implement it in a way that makes sense for others.

Maybe you could give me some pseudocode how you would ideally like to use behavioural cloning and/or GAIL in real life?

ONNX models is not on my crosshairs right now thanks for bringing to my my attention!
